lib/tapioca/gem/listeners/source_location.rb

```diff
@@ -66,6 +66,10 @@ def add_source_location_comment(node, file, line)
             return
           end
 
+          # Sanitize the gem version from the comment to avoid excessive churn in the RBI when upgrading gems. The
+          # version is always between a hyphen at the end of the gem name and the `/lib` folder
```

Comment:
Maybe we can also mention that having `-VERSION` will make `ruby-lsp` easier to substitute it with the actual version for path restoration?
I think currently it's not clear on why we keep this if the reader doesn't know how we'll use the source comment.